azurelinuxagent/ga/exthandlers.py

```diff
@@ -290,10 +290,13 @@ def run(self):
 
             if self._extension_processing_allowed() and self._incarnation_changed(etag):
                 activity_id, correlation_id = self.get_activity_and_correlation_id()
-                logger.info(
-                    "ProcessGoalState started [incarnation {0}; Activity Id: {1}; Correlation Id: {2}]".format(etag,
-                                                                                                               activity_id,
-                                                                                                               correlation_id))
+
+                log_msgs = ["Incarnation: {0}".format(etag)]
+                if activity_id is not None:
```

Comment:
You added this handling because the feature to add activityid/correlationid is not released everywhere yet.
What if, when this is released, either value is None due to a bug in the xml or in our parsing logic? Wouldn't we want to log it to surface the issue? What is the harm of always logging the ids regardless of their values? They will be None for the time being, but this will change as that feature is released.

Reply:
Hmm good point. I added this because there's no ETA on the release in CRP and I didnt want the log messages to look dirty. 

This is how it would look if those values are None - 
`ProcessGoalState completed [Incarnation: 1; 10 ms; Activity Id: ; Correlation Id: ]`

This looks very dirty. 
I can either make them say 'NA' or something or just not add them at all. 

I do like your point though. Let me replace the empty spaces with NA or something to make it more apparent

Reply:
Yeah, basically handling a non-value would do it. Thanks!